pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_cluster_settings failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_cluster_settings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/8626728?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/8626728?buildTab=artifacts#/) on master @ [ff67a4ba86d710db090ce700f229020365851183](https://github.com/cockroachdb/cockroach/commits/ff67a4ba86d710db090ce700f229020365851183):


```
[06:30:09] ALTER TENANT ALL RESET CLUSTER SETTING server.mem_profile.total_dump_size_limit;
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3node-tenant-default-configs (no issue given)
[06:30:09] SHOW CLUSTER SETTING server.mem_profile.total_dump_size_limit FOR TENANT [0];
[06:30:09] 	-- XFAIL;
[06:30:09] SHOW CLUSTER SETTING server.mem_profile.total_dump_size_limit FOR TENANT [1];
[06:30:09] 	-- XFAIL;
[06:30:09] SHOW CLUSTER SETTING server.mem_profile.total_dump_size_limit FOR TENANT [1111];
[06:30:09] 	-- XFAIL;
[06:30:09] SHOW CLUSTER SETTING server.mem_profile.total_dump_size_limit FOR TENANT [10];
[06:30:09] 	-- OK;
[06:30:09] SHOW CLUSTER SETTING server.mem_profile.total_dump_size_limit FOR TENANT "tenant-10";
[06:30:09] 	-- OK;
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3snode-tenant-default-config (no issue given)
[06:30:09] SHOW CLUSTER SETTINGS FOR TENANT [0];
[06:30:09] 	-- XFAIL;
[06:30:09] SHOW CLUSTER SETTINGS FOR TENANT [1];
[06:30:10] 	-- XFAIL;
[06:30:10] SHOW CLUSTER SETTINGS FOR TENANT [1111];
[06:30:10] 	-- XFAIL;
[06:30:10] SELECT variable, type, origin FROM [SHOW CLUSTER SETTINGS FOR TENANT [10]]
WHERE variable IN ('jobs.scheduler.enabled', 'jobs.retention_time') ORDER BY 1;
[06:30:10] 	-- OK;
[06:30:10] SELECT variable, type, origin FROM [SHOW CLUSTER SETTINGS FOR TENANT "tenant-10"]
WHERE variable IN ('jobs.scheduler.enabled', 'jobs.retention_time') ORDER BY 1;
[06:30:10] 	-- OK;
[06:30:10] SELECT variable, type, public, origin FROM [SHOW ALL CLUSTER SETTINGS FOR TENANT [10]]
WHERE variable IN ('jobs.scheduler.enabled', 'jobs.retention_time') ORDER BY 1;
[06:30:10] 	-- OK;
[06:30:10] SELECT variable, type, public, origin FROM [SHOW ALL CLUSTER SETTINGS FOR TENANT "tenant-10"]
WHERE variable IN ('jobs.scheduler.enabled', 'jobs.retention_time') ORDER BY 1;
[06:30:10] 	-- OK;
[06:30:10] SHOW CLUSTER SETTING sql.defaults.distsql;;
[06:30:10] 	-- FAIL
    logic.go:2892: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3738/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:331: SHOW CLUSTER SETTING sql.defaults.distsql;
        expected:
            NOTICE: using global default sql.defaults.distsql is not recommended
            HINT: use the ``ALTER ROLE ... SET`` syntax to control session variable defaults at a finer-grained level. See: https://www.cockroachlabs.com/docs/v23.1/alter-role.html#set-default-session-variable-values-for-a-role
            
        but found (query options: "noticetrace") :
            NOTICE: using global default sql.defaults.distsql is not recommended
            HINT: use the ``ALTER ROLE ... SET`` syntax to control session variable defaults at a finer-grained level. See: https://www.cockroachlabs.com/docs/dev/alter-role.html#set-default-session-variable-values-for-a-role
[06:30:10] 	-- FAIL
    logic.go:2151: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3738/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:337: too many errors encountered, skipping the rest of the input
[06:30:10] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3738/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings with config fakedist-disk: 71 tests, 2 failures
[06:30:10] --- total progress: 71 statements
--- total: 71 tests, 2 failures
    --- FAIL: TestLogic_cluster_settings/tenant-cluster-settings (0.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_cluster_settings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-24324

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_cluster_settings [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8629730?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/8629730?buildTab=artifacts#/) on master @ [d4f63d7e7f77a60819ce3fcec14b173e3e6acac8](https://github.com/cockroachdb/cockroach/commits/d4f63d7e7f77a60819ce3fcec14b173e3e6acac8):


```
=== RUN   TestLogic_cluster_settings
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/0205c25cd2c9009394cc81369c4dd2f5/logTestLogic_cluster_settings1009032215
    test_log_scope.go:79: use -show-logs to present logs inline
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3node-tenant-default-configs (no issue given)
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3node-tenant-default-configs (no issue given)
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3node-tenant-default-configs (no issue given)
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3node-tenant-default-configs (no issue given)
=== CONT  TestLogic_cluster_settings
    logic.go:4155: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0205c25cd2c9009394cc81369c4dd2f5/logTestLogic_cluster_settings1009032215
--- FAIL: TestLogic_cluster_settings (1.75s)
=== RUN   TestLogic_cluster_settings/tenant-cluster-settings
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3node-tenant-default-configs (no issue given)
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3node-tenant-default-configs (no issue given)
    logictestbase.go:512: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports 3snode-tenant-default-config (no issue given)
    logic.go:2892: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:331: SHOW CLUSTER SETTING sql.defaults.distsql;
        expected:
            NOTICE: using global default sql.defaults.distsql is not recommended
            HINT: use the ``ALTER ROLE ... SET`` syntax to control session variable defaults at a finer-grained level. See: https://www.cockroachlabs.com/docs/v23.1/alter-role.html#set-default-session-variable-values-for-a-role
            
        but found (query options: "noticetrace") :
            NOTICE: using global default sql.defaults.distsql is not recommended
            HINT: use the ``ALTER ROLE ... SET`` syntax to control session variable defaults at a finer-grained level. See: https://www.cockroachlabs.com/docs/dev/alter-role.html#set-default-session-variable-values-for-a-role
    logic.go:2151: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings:337: too many errors encountered, skipping the rest of the input
[10:03:09] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2805/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/cluster_settings with config fakedist-disk: 71 tests, 2 failures
--- total: 71 tests, 2 failures
    --- FAIL: TestLogic_cluster_settings/tenant-cluster-settings (0.18s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_cluster_settings.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/cluster_settings
+++ b/pkg/sql/logictest/testdata/logic_test/cluster_settings
@@ -327,11 +327,8 @@ WHERE variable IN ('jobs.scheduler.enabled', 'jobs.retention_time') ORDER BY 1
 jobs.retention_time     d  true   no-override
 jobs.scheduler.enabled  b  false  no-override
 
-query T noticetrace
+statement notice NOTICE: using global default sql.defaults.distsql is not recommended\nHINT: use the `ALTER ROLE ... SET` syntax to control session variable defaults at a finer-grained level. See: https://www.cockroachlabs.com/docs/.*/alter-role.html#set-default-session-variable-values-for-a-role
 SHOW CLUSTER SETTING sql.defaults.distsql;
-----
-NOTICE: using global default sql.defaults.distsql is not recommended
-HINT: use the `ALTER ROLE ... SET` syntax to control session variable defaults at a finer-grained level. See: https://www.cockroachlabs.com/docs/v23.1/alter-role.html#set-default-session-variable-values-for-a-role
 
 query T noticetrace
 SHOW CLUSTER SETTING sql.notices.enabled

